hash_practice/exercises.py

```diff
@@ -1,29 +1,32 @@
+from collections import defaultdict
 
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
     """
-    pass
+    anagrams = []
+    word_dict = anagrams_helper(strings)
+    for word in word_dict.values():
+        anagrams.append(word)
+    return anagrams
 
-def top_k_frequent_elements(nums, k):
-    """ This method will return the k most common elements
-        In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
-    """
-    pass
 
+def top_k_frequent_elements(nums, k):
+    """ This method will return the k most common elements"""
+    
+    num_frequency = {}
+    for num in nums:
+        if not num in num_frequency.keys():
+            num_frequency[num] = 1
+        num_frequency[num] += 1	
+    k_most_frequent_keys = sorted(num_frequency, key=num_frequency.get, reverse=True)[:k]
+    return k_most_frequent_keys
 
-def valid_sudoku(table):
-    """ This method will return the true if the table is still
-        a valid sudoku table.
-        Each element can either be a ".", or a digit 1-9
-        The same digit cannot appear twice or more in the same 
-        row, column or 3x3 subgrid
-        Time Complexity: ?
-        Space Complexity: ?
-    """
-    pass
 
+# anagram_helper
+def anagrams_helper(source_list):

```

Comment:
Minor style thing: in general, you should try to keep helper functions like this close to the functions that use them just so a reader doesn't have to hop around the source code.

Sometimes that's not possible, but it usually is.